### PR TITLE
fix: copy to clipboard error

### PR DESCRIPTION
**Pull request by iamsriranga**
## What kind of change does this PR introduce?

Bug fix

## What is the current behavior?

issue https://github.com/supabase/supabase/issues/14090

## Additional context

Add any other context or screenshots.

### CODE REVIEW

**Comment by saideepesh000** [comment]
This might fix it, but not sure how to test it

**Comment by saltcod** [comment]
Thanks @saideepesh000!   Would you mind outlining how you tested this one? Which browsers, which pages in the dashboard, etc. Appreciated!

**Comment by Ziinc** [comment]
This PR touches on this ticket for [copy button bug](https://www.notion.so/supabase/Logs-Explorer-cannot-copy-paste-from-log-panel-255f741002734000bbe98e36aaa404b1?pvs=4) in logs ui. In actuality this `copyToClipboard` bug **affects copying in studio for all firefox browsers due to lack of `window.ClipboardItem`**

https://caniuse.com/mdn-api_clipboarditem

there should be a fallback to pass the text directly to `navigator.clipboard.writeText`, which is [supported by all browsers](https://caniuse.com/mdn-api_navigator_clipboard), as demonstrated [here](https://stackoverflow.com/questions/400212/how-do-i-copy-to-the-clipboard-in-javascript).

Furthermore, since we are simply passing text to the clipboard, is the use of ClipboardItem really necessary? looks like we cna use `navigator.clipboard.writeText` and simplify the logic

https://www.w3.org/TR/clipboard-apis/#dom-clipboard-writetext

File: studio/lib/helpers.ts [hunk 1]
```diff
@@ -162,10 +162,11 @@ export const copyToClipboard = (str: string | Promise<string>, callback = () =>
   const focused = window.document.hasFocus()
   if (focused) {
     if (window.ClipboardItem) {
```
Comment (Ziinc) [inline studio/lib/helpers.ts:164]: remove this check with a check for window.navigator.clipboard.

File: studio/lib/helpers.ts [hunk 2]
```diff
@@ -162,10 +162,11 @@ export const copyToClipboard = (str: string | Promise<string>, callback = () =>
...
-      const text = new ClipboardItem({
-        'text/plain': Promise.resolve(str).then((text) => new Blob([text], { type: 'text/plain' })),
-      })
-      window.navigator?.clipboard?.write([text]).then(callback)
+      Promise.resolve(str)
```
Comment (Ziinc) [inline studio/lib/helpers.ts:165]: replace with `writeText`

Comment (Ziinc) [reply studio/lib/helpers.ts:165]: please use this as reference https://www.w3.org/TR/clipboard-apis/#dom-clipboard-writetext

Comment (Ziinc) [reply studio/lib/helpers.ts:165]: looks like writeText is already used below, so this is completely redundant. can be removed.

**Review by Ziinc — DISMISSED** [review]

**Comment by Ziinc** [comment]
@saideepesh000 are you able to get the review changes in? otherwise i'll pick this up

File: studio/lib/helpers.ts [hunk 3]
```diff
@@ -158,21 +158,22 @@ export const snakeToCamel = (str: string) =>
...
-      })
-      window.navigator?.clipboard?.write([text]).then(callback)
-
-      return
+    if (navigator.clipboard && typeof navigator.clipboard.writeText === 'function') {
+      try {
+        await navigator.clipboard.writeText(str)
+        callback()
+      } catch (error) {
+        console.warn('Unable to copy to clipboard')
```
[the remark this answers is not in the document]
Comment (saltcod) [reply studio/lib/helpers.ts:172]: Feels like a silent fail is fine for this one? wdyt?

[the remark this answers is not in the document]
Comment (Ziinc) [reply studio/lib/helpers.ts:172]: Silent fail is fine since we aren't supporting older browsers.

**Review by gregnr — APPROVED** [review]

**Pull request merged**